src/lamp_py/bus_performance_manager/vehicle_events.py

```diff
@@ -0,0 +1,166 @@
+from datetime import timedelta, date
+from typing import Optional
+import re
+
+import polars as pl
+
+from lamp_py.aws.s3 import (
+    file_list_from_s3_with_details,
+    get_datetime_from_partition_path,
+    get_last_modified_object,
+)
+
+from lamp_py.runtime_utils.remote_files import RemoteFileLocations
+
+
+def get_new_event_files() -> pl.DataFrame:
```

Comment:
suggestion: Returning a dataframe here creates a bit of a black box with regards to our CI tools. Something like a named tuple of dictionaries could help with using the output of this function by downstream parts of the application.